server/tests/src/main/scala/sttp/tapir/server/tests/ServerMultipartTests.scala

```diff
@@ -58,6 +63,41 @@ class ServerMultipartTests[F[_], ROUTE](
             r.body should include regex "name=\"data\"[\\s\\S]*oiram hcaep"
           }
       },
+      testServer(in_file_list_multipart_out_multipart) { (mfu: MultipleFileUpload) =>
+        val files = mfu.files.map { part =>
+          Part(
+            part.name,
+            writeToFile(Await.result(readFromFile(part.body), 3.seconds) + " result"),
+            part.otherDispositionParams,
+            Nil
+          ).header("X-Auth", part.headers.find(_.is("X-Auth")).map(_.value + "x").getOrElse(""))
+        }
+        pureResult(MultipleFileUpload(files).asRight[Unit])
+      } { (backend, baseUri) =>
+        val file1 = writeToFile("peach mario 1")
+        val file2 = writeToFile("peach mario 2")
+        val file3 = writeToFile("peach mario 3")
+        basicStringRequest
+          .post(uri"$baseUri/api/echo/multipart")
+          .multipartBody(
+            multipartFile("files", file1).fileName("fruit-data-1.txt").header("X-Auth", "12Aa"),
+            multipartFile("files", file2).fileName("fruit-data-2.txt").header("X-Auth", "12Ab"),
+            multipartFile("files", file3).fileName("fruit-data-3.txt").header("X-Auth", "12Ac")
+          )
+          .send(backend)
+          .map { r =>
+            r.code shouldBe StatusCode.Ok
+            println(r.body)
+            if (partOtherHeaderSupport) {
+              r.body should include regex "X-Auth: 12Aax"
```

Comment:
wouldn't simple `should include ("xyz")` work here? or is there a regex to check?

Reply:
fixed